src/external/VTK.jl

```diff
@@ -0,0 +1,58 @@
+import WriteVTK: vtk_grid, vtk_save
+
+# Uses WriteVTK.jl to convert scfres structure to VTk file format
+
+"""
+    save_scfres(filename, scfres, Val(:vtk))
+
+The function takes in the VTK filename and the scfres structure and stores into a VTK file.
+
+Grid Values:
+
+- ρ -> Density in real space
+- ψreal\\_k(i)\\_band(j) -> Real values of Bloch waves in real space
+- ψimag\\_k(i)\\_band(j) -> Imaginary values of Bloch waves in real space
+- ρspin -> Real value of ρspin are stored if ρspin in present
+
+MetaData:
+Energy , EigenValues, Fermi Level and occupation.
+"""
+function save_scfres(filename::AbstractString, scfres::NamedTuple, format::Val{Symbol(".vts")})
+    # Initialzing the VTK Grid
+    basis = scfres.basis
+    grid = zeros(3, basis.fft_size...)
+    for (idcs, r) in zip(CartesianIndices(basis.fft_size), r_vectors_cart(basis))
+        grid[:, Tuple(idcs)...] = r
+    end
+    vtkfile = vtk_grid(filename, grid)
+    
+    # Storing the Bloch Waves in Real space
+    for i in 1:length(basis.kpoints)
+        for j in 1:size(scfres.ψ[1])[2]
+            vtkfile["ψ_real_k$(i)_band$(j)"] = real.(G_to_r(basis, basis.kpoints[i], scfres.ψ[i][:,j]))
+            vtkfile["ψ_imag_k$(i)_band$(j)"] = imag.(G_to_r(basis, basis.kpoints[i], scfres.ψ[i][:,j]))
```

Comment:
Much better. Now can you use iband and ik as iteration variables (a convention we use throughout the code) and also assign the G_to_r expression to a variable since this avoids computing it twice (quite costly) and also shortens the lines below the limit.